sycl/doc/design/KernelFusionJIT.md

```diff
@@ -158,7 +158,115 @@ The metadata is attached to a function that will become the fused kernel:
+- All the fused nd-ranges must have the same offset.
+
+As we can see, there is no restrictions in the number of dimensions or global sizes of the input nd-ranges.
+
+#### Work-item remapping
+
+Work-item remapping is performed at the input kernel level, i.e., a different remapping is performed for each input kernel, as different input nd-ranges will result in different remappings.
+
+This remapping consists on an inter-procedural pass replacing each built-in querying components of a work-item, e.g., the global id or the local size, with a JIT-generated value.
+
+First of all, work-item remapping will always be performed when the list of input nd-ranges is heterogeneous. Additional remapping conditions are present for the following work-item components. For each input kernel:
+
+- `num_work_groups` and `local_size`: Only performed if the input nd-range has an explicit local size, may result in better performance, as this replaces built-in calls with constants;
+- `global_id`, `local_id` and `group_id`: Only needed if the number of dimensions differ w.r.t. that of the fused kernel or any component of the global size in the range [2, N] differs.
+
+Once this rules are set, also taking into account remapping constraints, the remapping is performed as follows for each input kernel:
+
+- `global_id`:
+  - `global_id(0) = GID / (global_size(1) * global_size(2))`
+  - `global_id(1) = (GID / global_size(2)) % global_size(1)`
+  - `global_id(2) = GID % global_size(2)`
+- `local_id`:
+  - `local_id(x) = global_id(x) % local_size(x)`
+- `group_id`:
+  - `group_id(x) = global_id(x) / local_size(x)`
+- `num_work_groups`:
+  - `num_work_groups(x) = global_size(x) / local_size(x)`
+- `global_size`:
+  - `global_size(x) = GS(x)`
+- `local_size`:
+  - `local_size(x) = LS(x)`
+- `global_offset`:
+  - `global_offset(x) = GO(x)`
+
+On the RHS of the expressions, component names refer to the remapped values and upper case `GS`, `LS` and `GO` values refer to each of the components of the original nd-range (global size, local size and global offset), whereas `GID` refers to the global id, which is an invariant during the fusion process.
+
+Special care needs to be taken when handling elements from the original nd-range, as the input index needs to be remapped to take into account different array subscript ordering of the underlying API w.r.t. SYCL. See [SYCL 2020 rev. 7 C.7.7](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:opencl:kernel-conventions-sycl) for more information on this index remapping.
+
+**Note**: As there is no `global_id` counterpart for PTX, global id is specified as `global_id(i) = group_id(i) * local_size(i) + local_id(i) + global_offset(i)`. This way, when targetting PTX, `local_size`, `local_id` and `group_id` will need to be remapped even when no explicit local size is provided. Remapping will take place as follows (also respecting original constraints):
```

Comment:
Emphasise that the remapping (as outlined) is *only* happening in this situation, i.e. when no local size is given. Otherwise, the same scheme as for SPIRV is used.